rust/src/x509/mod.rs

```diff
@@ -46,6 +48,18 @@ pub enum X509DecodeError {
 
 pub struct X509(X509Certificate<'static>);
 
+pub struct SCGeneralName<'a>(&'a GeneralName<'a>);
+
+impl<'a> fmt::Display for SCGeneralName<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut fin_str = format!("{}", self.0);
+        fin_str = fin_str[8..].to_string();
+        let fin_str_len = fin_str.len() - 1;
+        fin_str = fin_str[..fin_str_len].to_string();
+        write!(f, "{}", fin_str)
```

Comment:
Can this be done in a simpler way? I couldn't find a definition to properly exploit in `generalnames.rs`

Reply:
>  Parsing the output to extract that seems fragile and could break with an upstream change.

Indeed. I could see it breaking if for some reason another GeneralName type was sent over. This is why I was unsure.

Thanks. Will try the approach as you suggested. :)